### PR TITLE
[Stackdriver] Samples for documentation. 

### PR DESCRIPTION
**Pull request by frankyn**
This PR is to provide samples for the Stackdriver documentation.

### CODE REVIEW

File: stackdriver/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Stackdriver Samples
+
+Samples used by Stackdriver documentation. The samples don't work out-of-the-box
```
Comment (remi) [inline stackdriver/README.md]: We don't need to point out that *"the samples don't work out-of-the-box"*. Let's remove that sentence.

File: stackdriver/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+and require some modification. The samples are provided as reference guides
+when using Rails, Sinatra, other-rack frameworks, and non-rack apps.
```
Comment (remi) [inline stackdriver/README.md]: > *"The samples are provided as reference guides for using Stackdriver with Rails, Sinatra, Rack, and pure Ruby"*

File: stackdriver/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+## Rails
+
+The `rails_configuration.rb` provides a configuration example that can be used
+in your `config/environments/*.rb` files.
+
+## Sinatra
+
+Sinatra exmple apps load middleware using a rack configuration file. For example
```
Comment (remi) [inline stackdriver/README.md]: Type: `exmple`

File: stackdriver/logging.ru [hunk 4]
```diff
@@ -0,0 +1,20 @@
+require_relative "sinatra_logging"
```
Comment (remi) [inline stackdriver/logging.ru]: Using rackup files for Sinatra examples adds confusion.

See [Sinatra: README](http://www.sinatrarb.com/intro.html). The intro uses a simple `myapp.rb` file.

In my experience with Sinatra, I *always* start with a simple .rb file. That is idiomatic Sinatra. Frequently `app.rb` which is run simple via `ruby app.rb`

Let's only use a rackup file for the Rack examples.

File: stackdriver/non_rack_debugger.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# [START explicit_debugger_norack]
```
Comment (remi) [inline stackdriver/non_rack_debugger.rb]: `non_rack_debugger.rb` --> `ruby_debugger.rb`

Let's not "non-rack". It's simply pure Ruby 😄

File: stackdriver/sinatra_error_reporting.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
+require "sinatra"
+
+get "/" do
+
```
Comment (remi) [inline stackdriver/sinatra_error_reporting.rb]: This is missing a response. At least respond with something

```ruby
get "/" do
  erb :template
end
```

Comment (remi) [reply stackdriver/sinatra_error_reporting.rb]: Actually, another example appears to use "hello world" so use that or make them consistent

File: stackdriver/Gemfile [hunk 7]
```diff
@@ -0,0 +1,7 @@
+source "https://rubygems.org"
+
+gem "sinatra", "~> 1.4"
```
Comment (remi) [inline stackdriver/Gemfile]: Y?

File: stackdriver/README.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+## Rails
+
+The `rails_configuration.rb` provides configuration examples that can be used
+in your `config/environments/*.rb` files.
+
+## Rack and Sinatra
+
+Sinatra example apps load middleware using a rack configuration file. Sinatra
+samples are prefixed with `sinatra_` and Rack examples use the extension `.ru`.
```
Comment (remi) [inline stackdriver/README.md]: This is so clear! README is really great 👍

File: stackdriver/error_reporting.ru [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+end
+# [END error_reporting_configure]
+
+# [START error_reporting_middleware]
+require "google/cloud/error_reporting"
+
+use Google::Cloud::ErrorReporting::Middleware
+# [END error_reporting_middleware]
+
+run Proc.new { |env| ["200", {"Content-Type" => "text/html"}, ["Hello world!"]] }
```
Comment (remi) [inline stackdriver/error_reporting.ru]: Most folks would use the dash rocket, `run -> { |env| ... }`

File: stackdriver/Gemfile (hunk 7 above)
Comment (frankyn) [reply stackdriver/Gemfile]: The integrations tests only cover 1.4 at the moment. I requested that Heng update google-cloud-ruby tests to include Sinatra 2.0 as it makes sense moving forward. 

Current tested versions:
https://github.com/GoogleCloudPlatform/google-cloud-ruby/tree/master/integration

Cur

**Comment by frankyn** [comment]
Thanks @remi I have updated the PR!

File: stackdriver/Gemfile (hunk 7 above)
Comment (frankyn) [reply stackdriver/Gemfile]: Tested the samples manually with Sinatra 2.0

**Pull request merged**